앱 내 이벤트 노출 팝업 추가

JuTaK97 opened the pull request:
- url로 이미지 가져오는 부분 미구현
- 서버와 통신 확인 필요

JuTaK97 committed:
팝업 기능 추가

- url로 이미지 가져오는 부분 미구현
- 서버와 통신 확인 필요

JuTaK97 committed:
lint 오류 테스트

https://github.com/pinterest/ktlint/issues/1452#issuecomment-1098875162
의 해결책 시도

sanggggg commented:
[sanggggg/refactor-example-pop-up](https://github.com/wafflestudio/SNUTT-android/tree/sanggggg/refactor-example-pop-up)
리팩토링 브랜치 참고 요망

JuTaK97 committed:
팝업 refactor 반영

https://github.com/wafflestudio/SNUTT-android/tree/sanggggg/refactor-example-pop-up

JuTaK97 committed:
헤더 추가

header에 x-os-type, x-os-version, x-app-version, x-app-type 추가

(https://github.com/wafflestudio/snutt/pull/201)

JuTaK97 committed:
팝업 수정

1. API 전송 시점을 RootActivity에서 HomeFragment로 넘어갈 때로 변경
2. API response 형식에 맞추어 변경 (PopupDto -> List<PopupDto>)
3. 버튼 터치 영역 확대
4. 로그인 및 회원 가입 후에도 팝업 출력

JuTaK97 committed:
팝업 수정

1. ApplicationModule에서 Inject하여 사용하는 형태로 변경
2. popup API 요청하는 경우는 (이미 로그인된 유저 && splash 화면 직후) || (로그인 이후)
3. 서버 응답으로 여러 개의 팝업이 올 경우, 하나씩 띄운다.

sanggggg commented on app/src/main/java/com/wafflestudio/snutt2/data/SNUTTStorage.kt:
```diff
@@ -57,6 +57,18 @@ class SNUTTStorage @Inject constructor(
         )
     )
 
+    val popupMap = PrefValue<Map<String, Long>>(
```
[nit]
shownPopUpIdsAndTimestamp

sanggggg commented on app/src/main/java/com/wafflestudio/snutt2/di/NetworkModule.kt:
```diff
@@ -51,6 +52,42 @@ object NetworkModule {
             }
+            .addInterceptor { chain ->
+                val newRequest = chain.request().newBuilder()
+                    .addHeader(
+                        "x-os-type",
+                        "android"
+                    )
+                    .build()
+                chain.proceed(newRequest)
+            }
+            .addInterceptor { chain ->
+                val newRequest = chain.request().newBuilder()
+                    .addHeader(
+                        "x-os-version",
+                        Build.VERSION.SDK_INT.toString()
+                    )
+                    .build()
+                chain.proceed(newRequest)
+            }
+            .addInterceptor { chain ->
+                val newRequest = chain.request().newBuilder()
+                    .addHeader(
+                        "x-app-version",
+                        BuildConfig.VERSION_NAME
+                    )
+                    .build()
+                chain.proceed(newRequest)
+            }
+            .addInterceptor { chain ->
+                val newRequest = chain.request().newBuilder()
+                    .addHeader(
+                        "x-app-type",
+                        if (BuildConfig.DEBUG) "debug" else "release"
+                    )
+                    .build()
```
하나의 addInterceptor 에서 처리하기?

JuTaK97 committed:
팝업 리뷰 반영

sanggggg committed:
Merge branch 'develop' into jutak97/pop_up
